mesh/local_peer.go

```diff
@@ -149,7 +149,10 @@ func (peer *LocalPeer) handleAddConnection(conn Connection) error {
 	}
 	_, isConnectedPeer := peer.router.Routes.Unicast(toName)
 	peer.addConnection(conn)
-	if isConnectedPeer {
+	if isRestartedPeer {
+		conn.Log("connection added (restarted peer)")
+		peer.router.SendAllGossipDown(conn)
+	} else if isConnectedPeer {
```

Comment:
also, you may want to track this as a separate issue, for release notes purposes, since it goes beyond DNS.